grails-bootstrap/src/main/groovy/grails/util/BuildSettings.groovy

```diff
@@ -257,6 +258,9 @@ class BuildSettings {
             }
         }
         BASE_DIR = System.getProperty(APP_BASE_DIR) ? new File(System.getProperty(APP_BASE_DIR)) :  ( IOUtils.findApplicationDirectoryFile() ?: new File("."))
+        GRAILS_APP_DIR_PRESENT = new File(BASE_DIR, "grails-app").exists() || new File(BASE_DIR, "Application.groovy").exists()
+        TARGET_DIR = new File(BASE_DIR, "build")
+        RESOURCES_DIR = !GRAILS_APP_DIR_PRESENT ? null : (System.getProperty(PROJECT_RESOURCES_DIR) ? new File(System.getProperty(PROJECT_RESOURCES_DIR)) : new File(TARGET_DIR, "resources/main"))
```

Comment:
These settings were moved here because even though BASE_DIR had the ability to change, their values were always based on its initial value.